File: app/src/main/java/io/ipoli/android/quest/data/SubQuest.java
package io.ipoli.android.quest.data;

import com.fasterxml.jackson.annotation.JsonIgnore;

import java.util.Date;

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 3/26/16.
 */
public class SubQuest {

    private String name;

    private Long completedAt;

    private Integer completedAtMinute;

    public SubQuest() {
    }

    public SubQuest(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @JsonIgnore
    public Date getCompletedAtDate() {
        return completedAt != null ? new Date(completedAt) : null;
    }

    @JsonIgnore
    public void setCompletedAtDate(Date completedAtDate) {
        completedAt = completedAtDate != null ? completedAtDate.getTime() : null;
    }

    public Long getCompletedAt() {
        return completedAt;
    }

    public void setCompletedAt(Long completedAt) {
        this.completedAt = completedAt;
    }

    public Integer getCompletedAtMinute() {
        return completedAtMinute;
    }

    public void setCompletedAtMinute(Integer completedAtMinute) {
        this.completedAtMinute = completedAtMinute;
    }

    @JsonIgnore
    public boolean isCompleted() {
        return getCompletedAtDate() != null;
    }
}

File: app/src/main/java/io/ipoli/android/quest/data/SubQuest.kt
package io.ipoli.android.quest.data

import io.realm.RealmObject
import java.util.*

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 8/19/17.
 */
open class SubQuest : RealmObject {

    var name: String? = null

    var completedAt: Long? = null

    var completedAtMinute: Int? = null

    constructor() {}

    constructor(name: String) {
        this.name = name
    }

    var completedAtDate: Date?
        get() = if (completedAt != null) Date(completedAt!!) else null
        set(completedAtDate) {
            completedAt = if (completedAtDate != null) completedAtDate.getTime() else null
        }

    val isCompleted: Boolean
        get() = completedAtDate != null
}